File: Programas/FrontEnd/src/types/Reserva.ts
import { DetalleReserva } from "./DetalleReserva";
import { Inquilino } from "./Inquilino";
import { TipoReserva } from "./TipoReserva";
import { EstadoReserva } from "./EstadoReserva";

export interface Reserva {
  reservaId?: number;
  detalleReserva: DetalleReserva[];
  inquilino: Inquilino;
  tipoReserva: TipoReserva;
  estadoReserva: EstadoReserva;
  seña: number;
  TotalAPagar: number;
  TotalPagado: number;
}

File: Programas/ApiReservaRes/WebApplication2333/Models/Reserva.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ApiReservaRes.Models
{
    public class Reserva
    {
        public int reservaId { get; set; }

        public List<DetalleReserva> detalleReserva { get; set; }


        public Inquilino inquilino { get; set; }
        public TipoReserva tipoReserva { get; set; }
        public EstadoReserva estadoReserva { get; set; }
        public decimal seña { get; set; }
        public decimal TotalAPagar { get; set; }
        public decimal TotalPagado { get; set; }
        public DateTime FechaRegistro { get; set; }

        public string descripcion { get; set; }


    }
}